cmd/upgrade.go

```diff
@@ -77,6 +81,7 @@ func init() {
 	UpgradeCmd.Flags().BoolP("help", "h", false, "Print this help message")
 	UpgradeCmd.Flags().StringArrayVar(&values, "set", []string{}, "set values on the command line (can specify multiple or separate values with commas: key1=val1,key2=val2)")
 	UpgradeCmd.Flags().String("image-registry", "", "Custom/Private docker image repository URL")
+	UpgradeCmd.Flags().StringVarP(&upgradeImageVariant, "image-variant", "", "default", "The image variant to use for the Dapr runtime, for example: mariner")
```

Comment:
interesting ... if this is an argument, how do you think this is going to interact with `--set global.tag=1.8.0-mariner`?
@yaron2 @ItalyPaleAle do you think we need this flag here? 
It does make it consistent across k8s init and upgrade ... but I believe there should be a precedence defined for one set by `--set` flag and one set by the `--image-variant` flag. WDYT?

Reply:
At last both are going to do the same thing i.e set global.tag.

Reply:
But which value should have precedence ? IMO it should be the separate flag that's defined and not the one through set ... but this has to be clearly defined and documented ...

Reply:
@ItalyPaleAle @yaron2 @artursouza WDYT?

Reply:
IMHO the CLI should not be trying to parse `--set`. If the user passes both `--set global.tag=1.8.0-mariner` and `--image-variant`, then both are passed to helm. `--set` is a lower-level option and users should know what they're doing.

Reply:
The values passed in `--runtime-version` and `--image-variant` take precedence over the values passed in `--set` flag. This looks fine as well since if we are explicitly passing runtime version, that should be prioritized